code/game/machinery/cryopod.dm

```diff
@@ -137,12 +141,23 @@
 	updateUsrDialog()
 	return
 
+/obj/machinery/computer/cryopod/proc/freeze_item(obj/item/I, preserve_status)
+	frozen_items += I
+	if(preserve_status == CRYO_OBJECTIVE)
+		objective_items += I
+	I.forceMove(src)
+	RegisterSignal(I, COMSIG_MOVABLE_MOVED, .proc/item_got_removed)
+
+/obj/machinery/computer/cryopod/proc/item_got_removed(obj/item/I)
+	objective_items -= I
+	frozen_items -= I
+	UnregisterSignal(I, COMSIG_MOVABLE_MOVED)
+
 /obj/machinery/computer/cryopod/proc/dispense_item(obj/item/I)
 	if(!(I in frozen_items))
 		return
 	I.forceMove(get_turf(src))
-	objective_items -= I
-	frozen_items -= I
+	item_got_removed(I)
```

Comment:
You don't need to call this directly, since `forceMove` calls `Moved` which sends the `COMSIG_MOVABLE_MOVED` signal which then calls the `item_got_removed` proc.
```suggestion
```

Reply:
Oh true that. Will remove it and add a comment to explain it